lti_consumer/outcomes.py

```diff
@@ -188,6 +192,28 @@ def handle_request(self, request):
             failure_values['imsx_description'] = "User not found."
             return response_xml_template.format(**failure_values)
 
+        # Sends an email when the user earns a grade if xblock setting "email_notifications" is enabled.
+        if self.xblock.email_notifications:
+
+            context = {
+                "username": real_user.username,
+                "user_first_name": real_user.first_name,
+                "lms_root": settings.LMS_ROOT_URL,
+                "platform_name": settings.PLATFORM_NAME,
+                "course_name": self.xblock.course.display_name,
+                "xblock_display_name": self.xblock.display_name,
+                "forum_link": settings.LMS_ROOT_URL + reverse('forum_form_discussion', args=[self.xblock.course_id]),
+                "xblock_url": settings.LMS_ROOT_URL + reverse(
+                    'jump_to', args=[self.xblock.course_id, self.xblock.location]
+                ),
+                "site_logo": '{root_url}/static/{site_theme}/images/lti_email_logo.png'.format(
+                    root_url=settings.LMS_ROOT_URL,
+                    site_theme=settings.DEFAULT_SITE_THEME,
+                ),
+            }
+
+            self.send_email(real_user.email, context)
```

Comment:
@arsentur 
Have you considered to use a background task for this?

Reply:
@cmltaWt0 @arsentur is too busy now and I as a free option have implemented your remark.  Thanks for it and please do a finale reviews I bet.